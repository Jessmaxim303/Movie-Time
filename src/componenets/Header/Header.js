import React, { useState } from 'react';
import Dropdown from '../Dropdown/Dropdown.js';
import './Header.scss';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { logout } from '../../actions';
import PropTypes from 'prop-types';
import netflix from '../../Assets/Netflix.png';
import bell from '../../Assets/Bell.svg';
import profile from '../../Assets/profile.svg';

export const Header = (props) => {

  let avgRating = props.ratings.reduce((acc, rating) => {
    return acc + rating.rating;
  }, 0) / props.ratings.length
	return (
    <article className="header_main-container">
      <Link to={'/'} className="header_main-link">
        <img className="netflix_logo" src={netflix}/>
      </Link>
      {
        !props.user
        ? <Link to={'/login'} className="header_log-button">Sign In</Link>
        : 
        <Dropdown data={props} rating={avgRating}/>
      }
    </article>
  )
}

export const mapStateToProps = state => ({
  ratings: state.ratings,
  user: state.user
});

export const mapDispatchToProps = dispatch => ({
  logout: () => dispatch( logout() )
})

export default connect(mapStateToProps, mapDispatchToProps)(Header);

Header.propTypes = {
  props: PropTypes.any,
  ratings: PropTypes.array,
  user: PropTypes.object
}
