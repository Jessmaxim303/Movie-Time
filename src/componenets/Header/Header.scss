.header_main-container {
	font-family: 'Nunito', sans-serif;
	background: #0a1612;
	color: #ffffff;
	border-bottom: 3px solid #008F11;
	background: linear-gradient(to right, #003306, black);
	display: grid;
	grid-template-columns: 33% 33% 33%;
	grid-template-areas: 'welcome-text title logoff'
}

.header_welcome-text {
	grid-area: welcome-text;
	margin-left: 5%;
}

.header_main-h1 {
	grid-area: title;
	align-self: center;
	justify-self: center;
	padding: 2%;
	font-size: 3rem;
	border-right: 3px solid #008F11;
	border-left: 3px solid #008F11;
}

.header_logoff-button {
	grid-area: logoff;
	justify-self: end;
	align-self: center;
	cursor: pointer;
	font-family: 'Nunito', sans-serif;
	font-size: 1.2rem;
	font-weight: 900;
	width: 30%;
	height: 30%;
	background: #008F11;
	color: #0a1612;
	border: none;
	border-radius: 4%;
}

.header_logoff-button:hover {
	transform: rotate(6deg);
	transition-duration: .3s;
}
