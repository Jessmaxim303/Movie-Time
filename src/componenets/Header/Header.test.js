import React from 'react';
import { shallow } from 'enzyme';
import { Header, mapStateToProps, mapDispatchToProps } from './Header.js';
import { logout } from '../../actions/index.js'


describe('Header', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Header
      ratings = {[{rating: 2}, {rating: 3}]}
       />)
  })
  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  describe('mapStateToProps', () => {
    it('should return an object with the user and ratings array', () => {
      const mockState = {
        movies: [{title: 'Parasite', rating: 9}, {title: '1917', rating: 8}],
        ratings: [{id: 25, rating: 9}, {id: 27, rating: 4}],
        user: {id: 123, name: 'me'},
        loadingStatus: true,
        someOtherProp: false
      }
      const expected = {
        ratings: [{id: 25, rating: 9}, {id: 27, rating: 4}],
        user: {id: 123, name: 'me'},
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })
  describe('mapDispatchToProps', () => {
    it('should call dispatch with a logout action when logout() is called', () => {
      const mockDispatch = jest.fn();
      const action = logout();
      const mappedProps = mapDispatchToProps(mockDispatch)

      mappedProps.logout();
      expect(mockDispatch).toHaveBeenCalledWith(action);
    })

  })
})
