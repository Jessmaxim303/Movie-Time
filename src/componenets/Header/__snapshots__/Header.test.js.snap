// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match a snapshot 1`] = `
<article
  className="header_main-container"
>
  <article
    className="header_welcome-text"
  >
    <h1>
      Welcome, [user]!
    </h1>
    <h3>
       Avg. Rating: 5.5 
    </h3>
    <h3>
      Number of Ratings:
    </h3>
  </article>
  <Link
    className="header_main-link"
    to="/"
  >
    Rancid Tomatillos
  </Link>
  <button
    className="header_logoff-button"
  >
    Log Off
  </button>
</article>
`;
